Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibs.cs

```diff
@@ -0,0 +1,1801 @@
+                _cachedTextFields = new TypeLibTextFields { _name = "_UserFormBase{unnamed}#" + parentUserFormUniqueId };
+            }
+
+            InitCommon();
+            DetectUserFormClass();
+        }
+
+        private bool _isDisposed;
+        public void Dispose()
+        {
+            if (_isDisposed) return;
+            _isDisposed = true;
+
+            _typeInfosWrapped?.Dispose();
+            _containerTypeLib?.Dispose();
+            _ITypeInfo_Aggregator?.Dispose();
+            _IVBEComponent_Aggregator?.Dispose();
+            _IVBETypeInfo_Aggregator?.Dispose();
+        }
+
+        private TypeLibTextFields? _cachedTextFields;
+        TypeLibTextFields CachedTextFields
+        {
+            get
+            {
+                if (!_cachedTextFields.HasValue)
+                {
+                    var cache = new TypeLibTextFields();
+                    target_ITypeInfo.GetDocumentation((int)TypeLibConsts.MEMBERID_NIL, out cache._name, out cache._docString, out cache._helpContext, out cache._helpFile);
+                    _cachedTextFields = cache;
+                }
+                return _cachedTextFields.Value;
+            }
+        }
+
+        public string Name { get => CachedTextFields._name; }
+        public string DocString { get => CachedTextFields._docString; }
+        public int HelpContext { get => CachedTextFields._helpContext; }
+        public string HelpFile { get => CachedTextFields._helpFile; }
+
+        public string GetProgID() => (Container?.Name ?? "{unnamedlibrary}") + "." + CachedTextFields._name;
+
+        public Guid GUID { get => Attributes.guid; }
+        public TYPEKIND_VBE TypeKind { get => (TYPEKIND_VBE)Attributes.typekind; }
+        
+        public bool HasPredeclaredId { get => Attributes.wTypeFlags.HasFlag(ComTypes.TYPEFLAGS.TYPEFLAG_FPREDECLID); }
+        public ComTypes.TYPEFLAGS Flags { get => Attributes.wTypeFlags; }
+
+        private bool HasNoContainer() => _containerTypeLib == null;
+
+        /// <summary>
+        /// Silently compiles the individual VBE component (class/module etc)
+        /// </summary>
+        /// <returns>true if this module, plus any direct dependent modules compile successfully</returns>
+        public bool CompileComponent()
+        {
+            if (HasVBEExtensions)
+            {
+                try
+                {
+                    target_IVBEComponent.CompileComponent();
+                    return true;
+                }
+                catch (Exception e)
+                {
+                    if (e.HResult == (int)KnownComHResults.E_VBA_COMPILEERROR)
+                    {
+                        return false;
+                    }
+                    else
+                    {
+                        // this is more for debug purposes, as we can probably just return false in future.
+                        throw new ArgumentException("Unrecognised VBE compiler error: \n" + e.ToString());
```

Comment:
I think that for final version, we can wrap this line in a `#if DEBUG` and in the else branch, return false. That way, the production version won't blow up but the RD developers can get useful diagnostics.

Reply:
agreed.